synapse/rest/client/register.py

```diff
@@ -763,7 +763,10 @@ async def _create_registration_details(
         Returns:
              dictionary for response from /register
         """
-        result = {"user_id": user_id, "home_server": self.hs.hostname}
+        result: Dict[str, Union[str, int]] = {
```

Comment:
```suggestion
        result: JsonDict = {
```

May as well label it a `JsonDict` as that is what we're returning

Reply:
TIL :)